src/core/server/deprecations/deprecations_service.test.ts

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+/* eslint-disable dot-notation */
+import { DeprecationsService } from './deprecations_service';
+import { httpServiceMock } from '../http/http_service.mock';
+import { mockRouter } from '../http/router/router.mock';
+import { mockCoreContext } from '../core_context.mock';
+import { DeprecationsFactory } from './deprecations_factory';
+
+describe('DeprecationsService', () => {
+  const coreContext = mockCoreContext.create();
+  describe('#setup', () => {
+    const http = httpServiceMock.createInternalSetupContract();
+    const router = mockRouter.create();
+    http.createRouter.mockReturnValue(router);
+    const deprecationsCoreSetupDeps = { http };
+
+    it('registers routes', () => {
+      const deprecationsService = new DeprecationsService(coreContext);
+      deprecationsService.setup(deprecationsCoreSetupDeps);
+      // Registers correct base api path
+      expect(http.createRouter).toBeCalledWith('/api/deprecations');
+      // registers get route '/'
+      expect(router.get.mock.calls[0][0]).toEqual({ path: '/', validate: false });
+      expect(router.get.mock.calls[0][1]).toBeInstanceOf(Function);
+    });
+
+    it('calls registerConfigDeprecationsInfo', () => {
+      const deprecationsService = new DeprecationsService(coreContext);
+      const mockRegisterConfigDeprecationsInfo = jest.fn();
+      deprecationsService['registerConfigDeprecationsInfo'] = mockRegisterConfigDeprecationsInfo;
+      deprecationsService.setup(deprecationsCoreSetupDeps);
+      expect(mockRegisterConfigDeprecationsInfo).toBeCalledTimes(1);
+    });
+
+    it('returns setup contract', () => {
+      const deprecationsService = new DeprecationsService(coreContext);
+      const internalSetupContract = deprecationsService.setup(deprecationsCoreSetupDeps);
+      expect(internalSetupContract).toMatchInlineSnapshot(`
+        Object {
+          "getRegistry": [Function],
+        }
+      `);
```

Comment:
Not sure of the value of those non-behavioural tests, as this is supposed to be ensured by TS anyway?

Reply:
Yea it might be redundant. I dont like relying on TS in any way but maybe this is an overkill